### ISSUE TITLE
Reverse-Prox for Docker

### ISSUE THREAD

**Issue by henfri** [rockstor/rockstor-core#1121]
Hello,

this is a feature-request and especialy a request for comment.
When running more and more rockons, we will face the issue, that ports will be conflicting; especially port 80.

Wouldn't it be great to have several virtual servers?
admin.rockstor
plex.rockstor
owncloud.rockstor
...

If I see this correctly, this guy has done most of the job for us already:
http://oskarhane.com/nginx-as-a-reverse-proxy-in-front-of-your-docker-containers/

In short the idea:
Ngingx is used as reverse-proxy. The script automatically finds the ports exposed by the docker host and modifies the nginx config accordingly.

Other posts in this direction:
http://jasonwilder.com/blog/2014/03/25/automated-nginx-reverse-proxy-for-docker/
https://www.timofejew.com/moving-to-docker-nginx/

Greetings,
Hendrik

**Comment by milleraa** [comment]
Very very useful thing in conjunction with lets encrypt.

**Comment by kanecko** [comment]
I like your idea, because I am trying to achieve it right now :D
Even a Rockstor-approved how-to in this manner would be great.

**Comment by FroggyFlox** [comment]
Hi @kanecko ,

I'm unfortunately too short on time to detail the procedure, but I would recommend you to have a look at the Nginx ProxyManager Rock-On. It will do just that.

Having this as a "few clicks" options in Rockstor has been a long standing goal of mine and we have implemented a lot of what is needed to make that easier now (see rocknets, for instance). We still not quite there but it is still very high on the list.
